modules/RTC/TraceablePeerConnection.js

```diff
@@ -1123,117 +1130,80 @@ TraceablePeerConnection.prototype._removeRemoteTrack = function(toBeRemoved) {
+            if (Array.isArray(ssrcGroups)) {
+                for (const group of ssrcGroups) {
+                    group.ssrcs = group.ssrcs.split(' ').map(ssrcStr => parseInt(ssrcStr, 10));
+                    ssrcInfo.groups.push(group);
                 }
-            }
 
-            const simGroup = mLine.ssrcGroups.find(group => group.semantics === 'SIM');
+                const simGroup = ssrcGroups.find(group => group.semantics === 'SIM');
 
-            // Add a SIM group if its missing in the description (happens on Firefox).
-            if (!simGroup) {
-                const groupSsrcs = mLine.ssrcGroups.map(group => group.ssrcs[0]);
+                // Add a SIM group if its missing in the description (happens on Firefox).
+                if (this.isSpatialScalabilityOn() && !simGroup) {
+                    const groupSsrcs = ssrcGroups.map(group => group.ssrcs[0]);
 
-                groupsMap.get(groupSsrcs[0]).push({
-                    semantics: 'SIM',
-                    ssrcs: groupSsrcs
-                });
+                    ssrcInfo.groups.push({
+                        semantics: 'SIM',
+                        ssrcs: groupSsrcs
+                    });
+                }
             }
-        }
 
-        let ssrcs = mLine.ssrcs;
-
-        // Filter the ssrcs with 'cname' attribute.
-        ssrcs = ssrcs.filter(s => s.attribute === 'cname');
-
-        for (const ssrc of ssrcs) {
-            // Use the mediaType as key for the source map for unified plan clients since msids are not part of
-            // the standard and the unified plan SDPs do not have a proper msid attribute for the sources.
-            // Also the ssrcs for sources do not change for Unified plan clients since RTCRtpSender#replaceTrack is
-            // used for switching the tracks so it is safe to use the mediaType as the key for the TrackSSRCInfo map.
-            const key = `${mLine.type}-${index}`;
-            const ssrcNumber = ssrc.id;
-            let ssrcInfo = ssrcMap.get(key);
-
-            if (!ssrcInfo) {
-                ssrcInfo = {
-                    ssrcs: [],
-                    groups: [],
-                    msid: key
-                };
-                ssrcMap.set(key, ssrcInfo);
-            }
-            ssrcInfo.ssrcs.push(ssrcNumber);
+            ssrcMap.set(sourceName, ssrcInfo);
 
-            if (groupsMap.has(ssrcNumber)) {
-                const ssrcGroups = groupsMap.get(ssrcNumber);
+            const oldSsrcInfo = this.localSSRCs.get(localTrack.rtcId);
+            const oldSsrc = this._extractPrimarySSRC(oldSsrcInfo);
+            const newSsrc = this._extractPrimarySSRC(ssrcInfo);
 
-                for (const group of ssrcGroups) {
-                    ssrcInfo.groups.push(group);
-                }
+            if (oldSsrc !== newSsrc) {
+                oldSsrc && logger.error(`${this} Overwriting SSRC for track=${localTrack}] with ssrc=${newSsrc}`);
```

Comment:
What's the effect of this? If it's not kosher to do why are we doing it?

Reply:
It has no affect on functionality since we fixed it recently but it's good to know when this happens and was therefore  logging it as an error. Technically it shouldn't be happening but recently we saw this happen in Chrome 123 or so. I can change it to debug.